src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp

```diff
@@ -87,13 +87,20 @@ namespace
             return Model;
         }
 
-        virtual void prepare_inputs(void* data) const APPLESEED_OVERRIDE
+        virtual void prepare_inputs(
+            const ShadingPoint& shading_point,
+            void*               data) const APPLESEED_OVERRIDE
         {
             DipoleBSSRDFInputValues* values =
                 reinterpret_cast<DipoleBSSRDFInputValues*>(data);
 
             // Precompute the relative index of refraction.
-            values->m_eta = values->m_outside_ior / values->m_inside_ior;
+            const double outside_ior =
```

Comment:
Are you sure about the logic here?

For comparison, here is the logic in the specular BTDF:

    if (shading_point.is_entering())
    {
        values->m_from_ior =
            shading_point.get_ray().get_current_ior();
        values->m_to_ior = values->m_ior;
    }
    else
    {
        values->m_to_ior =
            shading_point.get_ray().get_previous_ior();
        values->m_from_ior = values->m_ior;
    }